File: clients/android/sample/src/androidTest/kotlin/dev/arizona/sample/TickerE2ETest.kt
package dev.arizona.sample

import androidx.compose.ui.test.junit4.createAndroidComposeRule
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.test.performClick
import org.junit.Rule
import org.junit.Test

/**
 * On-device server push: navigate to /native/ticker; a handle_info timer pushes
 * count updates unsolicited, so the displayed count rises without any tap. Run
 * with ./gradlew :sample:connectedCheck (server at 10.0.2.2:4040).
 */
class TickerE2ETest {
    @get:Rule
    val rule = createAndroidComposeRule<MainActivity>()

    @Test
    fun countRisesFromServerPush() {
        rule.waitForText("Ticker")
        rule.onNodeWithText("Ticker").performClick()
        // No tap: the server's timer drives the increments.
        rule.waitForText("Tick: 1")
        rule.waitForText("Tick: 2")
    }
}

File: clients/ios/Sample/UITests/TickerE2ETest.swift
import XCTest

/// On-device server push: navigate to /native/ticker; a handle_info timer pushes
/// count updates unsolicited, so the displayed count rises without any tap. The
/// Android analogue is TickerE2ETest.
final class TickerE2ETest: XCTestCase {
    private var app: XCUIApplication!

    override func setUpWithError() throws {
        continueAfterFailure = false
        app = launchSample()
    }

    func testCountRisesFromServerPush() {
        tapButton(app, "Ticker")
        // No tap: the server's timer drives the increments.
        waitForText(app, "Tick: 1")
        waitForText(app, "Tick: 2")
    }
}
